### ISSUE TITLE
Issue using nanoMQ behind reverse proxy ngnix

### ISSUE THREAD

**Issue by EnricoBeltramo** [nanomq/nanomq#1801]
**Describe the bug**
I have a working configuration with Mosquitto + a reverse proxy and I would replace it with nanomq.
My aim is forward the websocket to a sub path of ngnix
I configured nanomq in this way:

# MQTT listener
listeners.tcp = {
  bind = "0.0.0.0:1883"     # The listener binds to all network interfaces on port 1883
}

# WebSocket listener
listeners.ws {
  bind = "0.0.0.0:8000"			# Bind to all network interfaces on port 8083
}

# Log settings
log = {
  to = [file, console]                  # Destination of the logs
  level = "warn"                        # Log level
  dir = "/tmp"                          # Directory for log files
  file = "nanomq.log"                   # Filename for the log file
  rotation = {
    size = "10MB"                       # Maximum size of each log file
    count = 5                           # Maximum rotation count of log files
  }
}

# Allow anonymous connections
auth {
	allow_anonymous = true # Allow clients to connect without providing username/password
}

And the ngnix: 

    location /mqtt
    {
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Host $host;

        proxy_pass http://localbroker:8000;
    }

How said before, the ngnix configuration works fine for a similar configuration with Mosquitto, but not for nanoMQ. I tried to use directly ws without ngnix and works fine.

**Expected behavior**

Correct forward of mqtt websocket in subpath /mqtt

**Actual Behavior**
Describe what occurred.
The ngnix say they don't found the source and so doesn't forwrd connection.

**To Reproduce**
If possible include actual reproduction test code here.
Minimal C test cases are perferred.

**Environment Details**
 - NanoMQ version latest docker
 - Operating system and version Linux ubuntu 22.04
 - Compiler and language used N.A
 - testing scenario

**Client SDK**
If possible include the mqtt sdk you used to connect to nanomq
Minimal C test cases are perferred.
N.A

**Additional context**

**Comment by JaylinYu** [comment]
Theoratically speaking, on the networking level, nanomq is just like mosquitto, nothing special.
is this issue only with MQTT websocket, or TCP as well?

**Comment by EnricoBeltramo** [comment]
I tested only websocket at moment. I'm using docker and I'm using the minimal image in docker hub (i.e. emqx/nanomq:0.14.1). MAy be is necessary to use other versions with more features (like emqx/nanomq:0.14.1-slim or emqx/nanomq:0.14.1-full)?

**Comment by EnricoBeltramo** [comment]
No, same issue with full docker. May be the broker answer only to ws:// prefix and not to http:// as expected from ngnix?

**Comment by EnricoBeltramo** [comment]
i found, with ngnix it work only if I add a "/" to proxy_pass:

location /mqtt
{
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
    proxy_set_header Host $host;

    proxy_pass http://localbroker:8000/; # <--- add "/" here
}

I don't know if is expected or it's an error

**Comment by JaylinYu** [comment]
I dont know either. but your discovery is definitely super helpful for late comers

**Comment by geostonemarten** [comment]
> i found, with ngnix it work only if I 




> i found, with ngnix it work only if I add a "/" to proxy_pass:
> 
> location /mqtt 
> ```
> proxy_pass http://localbroker:8000/; # <--- add "/" here
> ```
> 

> I don't know if is expected or it's an error
Hi, that a humain error. 

The same question was asked about this on Stackoverflow and it contains the useful explanations. This topic also applies to Apache HTTPD
https://stackoverflow.com/questions/22759345/nginx-trailing-slash-in-proxy-pass-url

**Comment by kuolemaaa** [comment]
Any help for a Traefik 3 user?

Giving that traefik + mosquitto worked like a charm but not with nanomq

**Comment by JaylinYu** [comment]
The Path is must, 
`
listeners.ws {
  bind = "0.0.0.0:1884"	# Bind to all network interfaces on port 1884
}
`
is an invalid config. shall be `0.0.0.0:1884\xxxx`
I don't familiar with Traefik, but try  `http://localbroker:1884/xxxx/`
